src/main/java/io/appium/java_client/android/connection/ConnectionState.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android.connection;
+
+public class ConnectionState {
+    public static final int AIRPLANE_MODE_MASK = 0b001;
+    public static final int WIFI_MASK = 0b010;
+    public static final int DATA_MASK = 0b100;
+
+    private final long bitMask;
+
+    public long getBitMask() {
+        return bitMask;
+    }
+
+    public ConnectionState(long bitMask) {
+        this.bitMask = bitMask;
+    }
+
+    /**
+     * @return true if airplane mode is enabled.
+     */
+    public boolean isAirplaneModeEnabled() {
+        return (bitMask & AIRPLANE_MODE_MASK) != 0;
```

Comment:
It seems that all these bitwise operations can be simplified via ```java.utils.BitSet```

Reply:
do you find the current implementation not readable enough? I think BitSet usage might add an unnecessary overhead, since the actual count of bitwise operations here is pretty limited anyway

Reply:
ok, np. However from my point of view `bs.set(AIRPLANE_MODE_BIT, false)` looks more naturaly than `this.bitMask = bitMask & ~AIRPLANE_MODE_MASK;` (Maybe I just don't like bitwise logic :)